Using `node:` imports breaks "enforce-module-boundaries" linting when root project is configured

### Current Behavior

When importing built-in Node libraries using the [`node:` prefix](https://nodejs.org/docs/latest-v18.x/api/esm.html#node-imports), the lint rule `@nrwl/nx/enforce-module-boundaries` fails. 

In my client project (where I first noticed the error), I get the error "Buildable libraries cannot import or export from non-buildable libraries". In this case the repo has the traditional `apps`/`libs` structure, I have root-level scripts configured, and I do not have a `build` target in the root `package.json` (hence the root project is not considered buildable by Nx). 

In the sample repo, this manifests as a circular dependency. But the underlying cause is the same: the Node import is resolving to a lib of the root project, which is incorrect.

Example:
```typescript
import assert from 'node:assert'
import { randomUUID } from 'node:crypto'
```

This is similar to #14571, and in fact the `node:` prefix was mentioned in this [comment](https://github.com/nrwl/nx/issues/14571#issuecomment-1416427731). 

### Expected Behavior

Built-in Node libraries should be ignored by the lint rule, since they are not part of the workspace.

This was partially fixed in #14846. When the built-in libraries are imported without the prefix, everything works as expected. The behavior should be the same for either import style (with or without the `node:` prefix).

### GitHub Repo

https://github.com/michaelmarcus-ml/nx-node-tutorial

### Steps to Reproduce

1. Clone the sample repo. Note that `auth.ts` includes the following import:
```typescript
import { randomUUID } from 'node:crypto'
```
2. Run `nx test auth`. Observe that the test passes.
3. Run `nx lint auth`. Observe the error message: 
```
Linting "auth"...

/Users/foo/code/nx-node-tutorial/auth/src/lib/auth.ts
  1:1  error  Circular dependency between "auth" and "nx-node-tutorial" detected: auth -> nx-node-tutorial -> auth

Circular file chain:
- auth/src/lib/auth.ts
- src/main.ts  @nrwl/nx/enforce-module-boundaries
```
4. Change the first line of `auth.ts` to the following:
```typescript
import { randomUUID } from 'crypto'
```
5. Run `nx test auth`. Observe that the test still passes.
6. Run `nx lint auth`. Observe that linting now passes. 

### Nx Report

```shell
Node : 18.13.0
   OS   : darwin x64
   npm  : 8.19.3
   
   nx                      : 15.7.1
   @nrwl/jest              : 15.7.1
   @nrwl/linter            : 15.7.1
   @nrwl/workspace         : 15.7.1
   @nrwl/cli               : 15.7.1
   @nrwl/devkit            : 15.7.1
   @nrwl/eslint-plugin-nx  : 15.7.1
   @nrwl/express           : 15.7.1
   @nrwl/js                : 15.7.1
   @nrwl/node              : 15.7.1
   @nrwl/tao               : 15.7.1
   @nrwl/webpack           : 15.7.1
   typescript              : 4.9.5
```


### Failure Logs

```shell
[2023-02-16T22:18:38.979Z] /src/app/libs/shared/event-sourcing-core/src/lib/repository/EventStreamRepository.ts
[2023-02-16T22:18:38.979Z]   1:1  error  Buildable libraries cannot import or export from non-buildable libraries  @nrwl/nx/enforce-module-boundaries
[2023-02-16T22:18:38.979Z]   2:1  error  Buildable libraries cannot import or export from non-buildable libraries  @nrwl/nx/enforce-module-boundaries
```


### Additional Information

_No response_

--- a/packages/nx/src/utils/target-project-locator.ts
+++ b/packages/nx/src/utils/target-project-locator.ts
@@ -12,7 +12,10 @@ import {
 } from '../project-graph/utils/find-project-for-path';
 import { builtinModules } from 'module';
 
-const builtInModuleSet = new Set(builtinModules);
+const builtInModuleSet = new Set<string>([
+  ...builtinModules,
+  ...builtinModules.map((x) => `node:${x}`),
+]);
 
 export class TargetProjectLocator {
   private projectRootMappings = createProjectRootMappings(this.nodes);

